### PR TITLE
Add dev version

### PR DESCRIPTION
**Pull request by mlondschien**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* ~~Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mlondschien** [comment]
@conda-forge-admin, please rerender

**Comment by mlondschien** [comment]
@xhochy

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+{% set version = "0.4.2a20210604" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 9134f56b7a2f465420f17b5bb0dee16047a70f01029c996b7ab3f197de2d0779
+  git_url: https://github.com/airspeed-velocity/asv
```
Comment (xhochy) [inline recipe/meta.yaml:10]:
```suggestion
  url: https://github.com/airspeed-velocity/asv/archive/ef016e233cb9a0b19d517135104f49e0a3c380e9.tar.gz
  sha256: addtheshamanually
```

**Comment by xhochy** [comment]
Created a `dev` branch and changed the base.

**Comment by xhochy** [comment]
@mlondschien Probably better to revert to the  `git_url` here with a comment that you cannot easily build from a tarball. It is probably best if someone would migrate upstream to e.g. `setuptools_scm` for the versioning instead of using their custom code but it would be too much work to patch that for just the nightly build.

Can you also make a separate PR to master where you add us two as a maintainer (without increasing the build number)?

File: recipe/meta.yaml (hunk 1 above)
Comment (xhochy) [inline recipe/meta.yaml:10]:
```suggestion
  # asv's setup.py doesn't support building dev releases without access to the git information.
  git_url: https://github.com/airspeed-velocity/asv
```

**Comment by xhochy** [comment]
@conda-forge/asv This is ready for review. We also added ourselves as maintainers in #24

**Comment by CJ-Wright** [comment]
maintainer PR accepted please merge as you see fit!

**Comment by xhochy** [comment]
Thanks!

**Pull request merged**